> @git-hulk I built current head on my laptop. You can pull image `foxdalas/kvrocks:latest` `amd64/arm64.` Everything works in AWS EKS.

Thanks @foxdalas, are you using this Dockerfile, I got the below error when compiling.

```shell
include/jemalloc/internal/rtree.h:109:33: error: variably modified 'root' at file scope
  109 |         rtree_leaf_elm_t        root[1U << (RTREE_NSB/RTREE_HEIGHT)];
```

--- a/.github/workflows/daily-ci.yaml
+++ b/.github/workflows/daily-ci.yaml
@@ -23,6 +23,30 @@ on:
     - cron: '0 1 * * *'
 
 jobs:
+  build-on-ubuntu-docker:
+    name: Build Docker Image
+    runs-on: ubuntu-18.04
+    steps:
+
+      - name: Checkout Code Base
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 64
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Available platforms
+        run: echo ${{ steps.buildx.outputs.platforms }}
+
+      - name: Build Docker Image
+        run: |
+          docker buildx build --platform linux/amd64,linux/arm64 .
+
   build-on-ubuntu-without-jemalloc:
     strategy:
       matrix:

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -77,12 +77,19 @@ jobs:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Available platforms
+        run: echo ${{ steps.buildx.outputs.platforms }}
+
       - name: Build And Push Docker Image
         run: |
-          docker build -t kvrocks/kvrocks:$RELEASE_TAG .
-          docker tag kvrocks/kvrocks:$RELEASE_TAG kvrocks/kvrocks:latest
-          docker push kvrocks/kvrocks:$RELEASE_TAG
-          docker push kvrocks/kvrocks:latest
+          docker buildx build --platform linux/amd64,linux/arm64 --tag kvrocks/kvrocks:$RELEASE_TAG --tag kvrocks/kvrocks:latest .
 
   release-rpm-packages:
     name: Release Rpm Packages

--- a/Dockerfile
+++ b/Dockerfile
@@ -17,7 +17,7 @@
 
 FROM ubuntu:22.04 as build
 
-RUN apt update && apt install -y make git autoconf libtool g++
+RUN apt update && apt install -y make git autoconf libtool g++ libsnappy-dev
 WORKDIR /kvrocks
 
 COPY . .

